syne_tune/optimizer/schedulers/synchronous/dehb.py

```diff
@@ -0,0 +1,736 @@
+        is ignored, even if it still contains entries.
+    random_seed : int
+        Master random seed. Generators used in the scheduler or searcher are
+        seeded using `RandomSeedGenerator`. If not given, the master random
+        seed is drawn at random here.
+    max_resource_attr : str
+        Key name in config for fixed attribute containing the maximum resource.
+        If given, trials need not be stopped, which can run more efficiently.
+    resource_attr : str
+        Name of resource attribute in result's obtained via `on_trial_result`.
+        Note: The type of resource must be int.
+    mutation_factor : float, in (0, 1]
+        Factor F used in the rand/1 mutation operation of DE
+    crossover_probability : float, in (0, 1)
+        Probability p used in crossover operation (child entries are chosen
+        with probability p)
+    support_pause_resume : bool
+        If True, `_suggest` supports pause and resume in the first bracket. If
+        the objective supports checkpointing, this is made use of.
+        Note: The resumed trial still gets assigned a new trial_id, but it
+        starts from the earlier checkpoint.
+    """
+
+    MAX_RETRIES = 50
+
+    def __init__(
+        self,
+        config_space: dict,
+        rungs_first_bracket: List[Tuple[int, int]],
+        num_brackets_per_iteration: Optional[int] = None,
+        **kwargs,
+    ):
+        super().__init__(config_space)
+        self._create_internal(rungs_first_bracket, num_brackets_per_iteration, **kwargs)
+
+    def _create_internal(
+        self,
+        rungs_first_bracket: List[Tuple[int, int]],
+        num_brackets_per_iteration: Optional[int] = None,
+        **kwargs,
+    ):
+        # Check values and impute default values
+        assert_no_invalid_options(
+            kwargs, _ARGUMENT_KEYS, name="DifferentialEvolutionHyperbandScheduler"
+        )
+        kwargs = check_and_merge_defaults(
+            kwargs,
+            set(),
+            _DEFAULT_OPTIONS,
+            _CONSTRAINTS,
+            dict_name="scheduler_options",
+        )
+        self.metric = kwargs.get("metric")
+        assert self.metric is not None, (
+            "Argument 'metric' is mandatory. Pass the name of the metric "
+            + "reported by your training script, which you'd like to "
+            + "optimize, and use 'mode' to specify whether it should "
+            + "be minimized or maximized"
+        )
+        self.mode = kwargs["mode"]
+        self.max_resource_attr = kwargs.get("max_resource_attr")
+        self._resource_attr = kwargs["resource_attr"]
+        self.mutation_factor = kwargs["mutation_factor"]
+        self.crossover_probability = kwargs["crossover_probability"]
+        self._support_pause_resume = kwargs["support_pause_resume"]
+        # Generator for random seeds
+        random_seed = kwargs.get("random_seed")
+        if random_seed is None:
+            random_seed = np.random.randint(0, 2**32)
```

Comment:
`get` allows you to provide a default value as the second param: `get("random_seed", np.random.randint(0, 2**32))`.

Reply:
This is a tricky one. If kwargs contains 'random_seed', but the value is None, get returns None. It only returns the default if the key is not there.